### PR TITLE
Output JUnit reports for Jest tests

### PR DESCRIPTION
**Pull request by simonsinclair**
No Issue.

**Overall change:** Following @jamesdonoh's work with adding reporting to our e2e tests, this adds JUnit reporting to our _unit tests_. The JUnit reporter will output logs in machine readable format and can help us determine which tests are failing regularly or taking a long time to run.

**Code changes:**
- Install `jest-junit` reporter.
- Configure reporter.
  - Reports are saved to the (already existing) `./reports/` directory.
  - Reports are named uniquely and not overwritten.
- Enable reporter on Jest runs.
- On CI, pass [--ci](https://jestjs.io/docs/en/cli#--ci) to Jest, so that's its aware of it's environment.

**Notes:**
- `uniqueOutputName` expects a string. 🤷🏼‍♂️

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -1,6 +1,18 @@
 const { jestDirAlias } = require('./dirAlias');
 
 module.exports = {
+  reporters: [
+    'default',
+    [
+      'jest-junit',
+      {
+        suiteName: 'Jest Tests',
+        outputDirectory: 'reports',
```
Comment (jamesdonoh) [inline jest.config.js]: Maybe we should put these under `reports/jest` and then we can group the other results/reports under there too.

**Review by jamesdonoh — COMMENTED** [review]
What happens to old test reports from previous runs? Are they automatically cleared/replaced?

File: jest.config.js (hunk 1 above)
Comment (simonsinclair) [reply jest.config.js]: Yes, that's a good idea.

**Comment by simonsinclair** [comment]
> What happens to old test reports from previous runs? Are they automatically cleared/replaced?

The reports are uniquely named, so won't be replaced and aren't removed. From what I've read this is what the Jenkin's JUnit plugin expects. I think we will have to come to a decision about how often rotate reports, so that our data visualisation is as representative of the current suite as it can be.

**Review by HarveyPeachey — COMMENTED** [review]
Is there any way to output a date/timestamp as part of the file naming for when the report was generated? Atm it's just "junit" and then a string of characters: `junit-22c2e4f0-475d-11ea-9967-dba6c268bcd1.xml`.

**Comment by simonsinclair** [comment]
> Is there any way to output a date/timestamp as part of the file naming for when the report was generated? Atm it's just "junit" and then a string of characters: `junit-22c2e4f0-475d-11ea-9967-dba6c268bcd1.xml`.

That string is a unique hash assigned by the reporter itself. Other metadata, like the timestamp, is stored within the file itself. We could implement a custom naming solution, but I'm not sure that would provide much value, because these files aren't designed to be read by humans.

**Comment by HarveyPeachey** [comment]
> > Is there any way to output a date/timestamp as part of the file naming for when the report was generated? Atm it's just "junit" and then a string of characters: `junit-22c2e4f0-475d-11ea-9967-dba6c268bcd1.xml`.
> 
> That string is a unique hash assigned by the reporter itself. Other metadata, like the timestamp, is stored within the file itself. We could implement a custom naming solution, but I'm not sure that would provide much value, because these files aren't designed to be read by humans.

Ahh skipped over the machine-readable bit 😅. Also, do we have to keep in mind the file size of these reports? Just one is 418KB

**Comment by chris-hinds** [comment]
> > > Is there any way to output a date/timestamp as part of the file naming for when the report was generated? Atm it's just "junit" and then a string of characters: `junit-22c2e4f0-475d-11ea-9967-dba6c268bcd1.xml`.
> > 
> > 
> > That string is a unique hash assigned by the reporter itself. Other metadata, like the timestamp, is stored within the file itself. We could implement a custom naming solution, but I'm not sure that would provide much value, because these files aren't designed to be read by humans.
> 
> Ahh skipped over the machine-readable bit 😅. Also, do we have to keep in mind the file size of these reports? Just one is 418KB

Were not committing these files to git right? if so the file size shouldn't matter too much

**Comment by hindsc52** [comment]
> > > Is there any way to output a date/timestamp as part of the file naming for when the report was generated? Atm it's just "junit" and then a string of characters: `junit-22c2e4f0-475d-11ea-9967-dba6c268bcd1.xml`.
> > 
> > 
> > That string is a unique hash assigned by the reporter itself. Other metadata, like the timestamp, is stored within the file itself. We could implement a custom naming solution, but I'm not sure that would provide much value, because these files aren't designed to be read by humans.
> 
> Ahh skipped over the machine-readable bit 😅. Also, do we have to keep in mind the file size of these reports? Just one is 418KB

Were not committing these files to git right? if so the file size shouldn't matter too much

**Comment by simonsinclair** [comment]
> Do we have to keep in mind the file size of these reports? Just one is 418KB

That's a good question - I'm hoping @hindsc52 can shed a bit of light on this. I believe we might need to clear or rotate the reports at given intervals to avoid filling up the disk.

**Comment by chris-hinds** [comment]
> > Do we have to keep in mind the file size of these reports? Just one is 418KB
> 
> That's a good question - I'm hoping @hindsc52 can shed a bit of light on this. I believe we might need to clear the reports to avoid filling up the disk. I was chatting to @jamesdonoh about this earlier and we think the results are preserved with the job data, so it should be safe to do this.

The Jenkins workspace is cleared after a successful build (or so it should be) build artefacts are then kept for 2 days

**Review by chris-hinds — APPROVED** [review]
LGTM 👍

**Review by HarveyPeachey — APPROVED** [review]
LGTM

**Pull request merged**